tests/test_eval.py

```diff
@@ -1,168 +1,159 @@
-        # Marian args:
-        "--models",
-        test_data_dir.join("fake_model.npz"),
-    ]
-
-    result = subprocess.run(command, env=env, stderr=subprocess.PIPE, check=False)
-    fail_on_error(result)
-
-    run_common_assertions(test_data_dir)
-
-    # Marian is passed a certain set of arguments. This can start failing if the marian
-    # arguments are adjusted.
-    marian_decoder_args = json.loads(test_data_dir.load("marian-decoder.args.txt"))
-    assert marian_decoder_args == [
-        # fmt: off
-        "--config", test_data_dir.join("fake_config.yml"),
-        "--quiet",
-        "--quiet-translation",
-        "--log",    test_data_dir.join("artifacts/wmt09.log"),
-        "--models", test_data_dir.join("fake_model.npz"),
-        # fmt: on
-    ], "The marian arguments matched."
-
-    assert "0.4\n1.0\n" in test_data_dir.load("artifacts/wmt09.metrics")
-
-
-def test_eval_gpu_sh() -> None:
-    test_data_dir, dataset_prefix, env = shared_setup(
-        {
-            "GPUS": "4",
+    test_data_dir.create_zst("wmt09.ru.zst", ru_sample),
+    return test_data_dir
+
+
+def run_normal_eval_task(test_data_dir: DataDir, task_name: str) -> None:
+    run_task(
+        task_name=task_name,
+        script="pipeline/eval/eval-gpu.sh",
+        work_dir=test_data_dir.path,
+        fetches_dir=test_data_dir.path,
+        env={
+            # This is where the marian_decoder_args will get stored.
+            "TEST_ARTIFACTS": test_data_dir.path,
+            # Replace marian with the one in the fixtures path.
             "MARIAN": fixtures_path,
-            "WORKSPACE": "1024",
-        }
+            # This is included via the poetry install
+            "COMPRESSION_CMD": "zstd",
+        },
     )
 
-    command = [
-        "pipeline/eval/eval-gpu.sh",
-        test_data_dir.join("artifacts/wmt09"),  # artifacts_prefix
-        dataset_prefix,
-        "en",  # src
-        "ca",  # trg
-        test_data_dir.join("fake_config.yml"),
-        test_data_dir.join("fake_model.npz"),
-    ]
-
-    result = subprocess.run(command, env=env, stderr=subprocess.PIPE, check=False)
-    fail_on_error(result)
-
-    # Marian is passed a certain set of arguments. This can start failing if the marian
-    # arguments are adjusted.
-    marian_decoder_args = json.loads(test_data_dir.load("marian-decoder.args.txt"))
-    assert marian_decoder_args == [
-        # fmt: off
-        "--config", test_data_dir.join("fake_config.yml"),
-        "--quiet",
-        "--quiet-translation",
-        "--log", test_data_dir.join("artifacts/wmt09.log"),
-        '--workspace', '1024',
-        '--devices', '4',
-        "--models", test_data_dir.join("fake_model.npz"),
-        # fmt: on
-    ], "The marian arguments matched."
-
-    assert "0.4\n1.0\n" in test_data_dir.load("artifacts/wmt09.metrics")
-
-
-def test_eval_quantized_sh() -> None:
-    test_data_dir, dataset_prefix, env = shared_setup(
-        {
+
+class Assert:
```

Comment:
I don't know how pythonic this is, but I wanted to collect all the asserts into an interface, as they were confusing with the other utility functions and test functions.